packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -148,6 +155,15 @@ Future<int> runCommandAndStreamOutput(
   final StreamSubscription<String> stdoutSubscription = process.stdout
     .transform<String>(utf8.decoder)
     .transform<String>(const LineSplitter())
+    .map((String line) {
+      if (detachFilter != null && detachFilter.hasMatch(line) && !result.isCompleted) {
+        // Detach from the process, assuming it will eventually complete successfully.
+        // Output printed after detaching (incl. stdout and stderr) will still be
+        // processed by [filter] and [mapFunction].
+        result.complete(0);
```

Comment:
check if this has already completed firstt

Reply:
Yes, I do check on the line above.